### PR TITLE
fix: removed line height from all inputs

### PR DESCRIPTION
**Pull request by allroundexperts**
### Details

This PR removes line height from all text inputs.

### Fixed Issues

$ https://github.com/Expensify/App/issues/14799
$ https://github.com/Expensify/App/issues/14799#issuecomment-1437292161

### Tests

1. Open settings page.
2. Open security settings, then click on close account.
3. Type any text in the multiline input that appears.

- [x] Verify that no errors appear in the JS console

### Offline tests

Not applicable

### QA Steps

1. Open settings page.
2. Open security settings, then click on close account.
3. Type any text in the multiline input that appears.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/30054992/224181187-17357d8b-d4f3-4418-b820-30c2fbd896f9.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

With LH:
https://user-images.githubusercontent.com/30054992/224184220-dd0f6f72-4ecb-419d-851a-fd0cb6897429.mp4

Without LH:
https://user-images.githubusercontent.com/30054992/224184244-052f750d-4dca-49e1-bb23-1b406099092a.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

With LH:
https://user-images.githubusercontent.com/30054992/224184323-fef3c45a-6acf-4833-951b-ae8d846de4a1.mp4

Without LH:
https://user-images.githubusercontent.com/30054992/224184330-9cc0ef10-24ee-4ad0-be0b-e4da2caf7b20.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/30054992/224181252-4ee79c33-7c0b-4b75-9bef-ffbcc2da0171.mov

</details>

<details>
<summary>iOS</summary>

With LH:
https://user-images.githubusercontent.com/30054992/224184440-55a50942-d446-4c26-affc-f874d7dc506a.mp4

Without LH:
https://user-images.githubusercontent.com/30054992/224184445-eca40ce6-4360-434a-be61-c9aed3186a16.mp4

</details>

<details>
<summary>Android</summary>

With LH:
https://user-images.githubusercontent.com/30054992/224184503-cd146bb8-91b4-45a7-b360-ba090d28240b.mp4

Without LH:
https://user-images.githubusercontent.com/30054992/224184511-3baa391f-ddf0-4282-b3cd-cc6854918190.mp4

</details>

### CODE REVIEW

File: src/components/TextInput/BaseTextInput.js [hunk 1]
```diff
@@ -294,9 +294,7 @@ class BaseTextInput extends Component {
                                             this.props.prefixCharacter && StyleUtils.getPaddingLeft(this.state.prefixWidth + styles.pl1.paddingLeft),
                                             this.props.secureTextEntry && styles.secureInput,
```
[the remark this answers is not in the document]
Comment (allroundexperts) [reply src/components/TextInput/BaseTextInput.js]: Fixed.

**Comment by allroundexperts** [comment]
@aimane-chnaif Updated.

File: src/styles/styles.js [hunk 2]
```diff
@@ -1400,6 +1401,9 @@ const styles = {
 
     // Be extremely careful when editing the compose styles, as it is easy to introduce regressions.
     // Make sure you run the following tests against any changes: #12669
```
[the remark this answers is not in the document]
Comment (allroundexperts) [reply src/styles/styles.js:1403]: @aimane-chnaif Android and iOS attached.

https://user-images.githubusercontent.com/30054992/224275917-49f09683-da12-4cdc-a8ab-7891e80c805a.mov

https://user-images.githubusercontent.com/30054992/224275971-54da7d9b-6932-466e-83c7-b6d2e82d102e.mov

File: src/components/TextInput/BaseTextInput.js [hunk 3]
```diff
@@ -293,10 +293,7 @@ class BaseTextInput extends Component {
                                             (!hasLabel || this.props.multiline) && styles.pv0,
                                             this.props.prefixCharacter && StyleUtils.getPaddingLeft(this.state.prefixWidth + styles.pl1.paddingLeft),
                                             this.props.secureTextEntry && styles.secureInput,
-
-                                            // Explicitly remove `lineHeight` from single line inputs so that long text doesn't disappear
-                                            // once it exceeds the input space (See https://github.com/Expensify/App/issues/13802)
```
[the remark this answers is not in the document]
Comment (allroundexperts) [reply src/components/TextInput/BaseTextInput.js:298]: Moved.

**Review by aimane-chnaif — APPROVED** [review]
@shawnborton @techievivek please check videos, especially in https://github.com/Expensify/App/pull/15804#issuecomment-1463681897.
I think no problem with changes and approving but will defer to you.

**Comment by shawnborton** [comment]
Hmm I'm gonna be honest, I don't love that now the composer box has a tighter line height - now the text you type doesn't match the same line height styling it would have when the text is sent and turned into a message. Is there a way to only do this for our text inputs and not the composer itself?

**Comment by allroundexperts** [comment]
@shawnborton @aimane-chnaif Just FYI, On android for the composer, there is a change in size of composer when you enter the first character of second line. This is also caused by the line height being set. Removing line height fixes that jump as well.

https://user-images.githubusercontent.com/30054992/224372070-96b7e4c4-d2f5-46f8-a6ef-960f0ec2dea9.mov

**Comment by shawnborton** [comment]
Interesting, got it. Okay, well I guess it makes sense to move forward with this and see how it feels then.

**Comment by allroundexperts** [comment]
> @allroundexperts can you also test #15864?

iOS:
Without LH:
Screenshot 2023-03-11 at 3 56 49 AM

With LH:
Screenshot 2023-03-11 at 3 55 53 AM

**Comment by shawnborton** [comment]
Hmmm the lines on iOS feel way too close together now. I'm starting to think that maybe this is not the best solution just to fix a cursor glitch in Android. Are we sure there isn't a better solution that can solve the Android issue without causing all of these other design inconsistencies?

**Comment by allroundexperts** [comment]
> > > @allroundexperts can you also test #15864?
> > 
> > 
> > iOS: With LH: Screenshot 2023-03-11 at 3 56 49 AM
> > Without LH: Screenshot 2023-03-11 at 3 55 53 AM
> 
> oops. too close. can you also test and share android diff?
@aimane-chnaif 

Screenshot 2023-03-11 at 4 06 39 AM

Screenshot 2023-03-11 at 4 07 21 AM

**Comment by shawnborton** [comment]
Can you share before/after on iOS with two lines of text (no emojis) please?

**Comment by allroundexperts** [comment]
> Hmmm the lines on iOS feel way too close together now. I'm starting to think that maybe this is not the best solution just to fix a cursor glitch in Android. Are we sure there isn't a better solution that can solve the Android issue without causing all of these other design inconsistencies?

@shawnborton We probably should not apply this to composer. There isn't any noticeable flickering in the composer anyways. For Close account multiline input, it looks fine because there are no emojis there.

**Comment by allroundexperts** [comment]
> Can you share before/after on iOS with two lines of text (no emojis) please?

Screenshot 2023-03-11 at 4 11 02 AM
Screenshot 2023-03-11 at 4 11 26 AM

**Comment by allroundexperts** [comment]
> > There isn't any noticeable flickering in the composer anyways
> 
> can you find root cause why barely noticeable in composer than other multiline inputs? I think those have same lineHeight, fontFamily and fontSize right now.
> 
> > it looks fine because there are no emojis there.
> 
> You still can input emoji though.
> 
> > Hmmm the lines on iOS feel way too close together now. I'm starting to think that maybe this is not the best solution just to fix a cursor glitch in Android. Are we sure there isn't a better solution that can solve the Android issue without causing all of these other design inconsistencies?
> 
> I agree fixing cursor glitch is not more important than too close between lines.

It's because initially composer is a `single line` input. It only becomes `multiline` once you go over to the next line. With other inputs, they are `multiline` since the beginning.

**Comment by shawnborton** [comment]
Which one is before or after in this [comment](https://github.com/Expensify/App/pull/15804#issuecomment-1464625448)?

**Comment by allroundexperts** [comment]
> Which one is before or after in this [comment](https://github.com/Expensify/App/pull/15804#issuecomment-1464625448)?

@shawnborton Small one is after the fix.

**Comment by shawnborton** [comment]
I kind of think we should go back to the drawing board here and try to find a better solution honestly. This one is starting to feel a bit hacky to me. Thoughts on that?

**Comment by allroundexperts** [comment]
@aimane-chnaif Thanks for letting me know. Do I get any compensation for this? Previously, I got paid if the PR was created but not merged due to this sort of issue. [Ref](https://github.com/Expensify/App/issues/14232)

**Comment by techievivek** [comment]
Hmmm... it looks like getting rid of line height is not a good choice over here. @allroundexperts, do you think you can explore some other solution here, or should we close this PR and look for a better fix?

**Comment by allroundexperts** [comment]
> Hmmm... it looks like getting rid of line height is not a good choice over here. @allroundexperts, do you think you can explore some other solution here, or should we close this PR and look for a better fix?

@techievivek The only other solution I can think of is fixing this on the native side. I'm not sure if I'll be able to do it since I don't have good knowledge of the native code. I'm still not sure why we can't just apply this to the close account text input on android only?

**Comment by techievivek** [comment]
@aimane-chnaif @shawnborton How about fixing this only for Android?

@allroundexperts has shared the screen recordings for Android.
With line-height:
https://user-images.githubusercontent.com/30054992/224184503-cd146bb8-91b4-45a7-b360-ba090d28240b.mp4

Without line-height:
https://user-images.githubusercontent.com/30054992/224184511-3baa391f-ddf0-4282-b3cd-cc6854918190.mp4

In case we don't feel confident with the above solution then I can reach out to Callstack developers and see if we can get anyone to work on this issue. Let me know what you guys think.

**Comment by techievivek** [comment]
@aimane-chnaif Good catch. It looks like this should get fixed after enabling the new architecture.

**Comment by techievivek** [comment]
Closing this PR as the callstack developers will be handling the linked issue now.

**Pull request closed without merging**